db/migrate/20161122031010_nullify_old_recovery_codes.rb

```diff
@@ -0,0 +1,16 @@
+class NullifyOldRecoveryCodes < ActiveRecord::Migration
+  class User < ActiveRecord::Base
+  end
+
+  def up
+    User.all.each do |user|
+      if user.recovery_code.present? && user.recovery_code =~ /^$2/
```

Comment:
ah you care correct. I shall have to hand in my Regex Club membership badge.

fixed.